pkg/mutation/path/tester/tester.go

```diff
@@ -71,16 +78,40 @@ func ValidatePathTests(location *parser.Path, pathTests []Test) error {
 
 // New creates a new Tester object
 func New(tests []Test) (*Tester, error) {
+	paths := make(map[int]*parser.Path)
```

Comment:
What is the relationship between the validations in `New` and `ValidatePathTests`?

Reply:
ValidatePathTests is called before New for non-metadata mutators. New does not accept a location, and a location is available at the construction of AssignMutator construction (where it calls New), but not inside Mutate of AssignMetadataMutator (where it calls New). This is a result of AssignMutator and AssignMetadataMutator caching different sets of data.

For this PR, I opted to go the smaller-PR route rather than also change how AssignMetadataMutator is constructed to make this more consistent. I'd rather that change be its own PR than expand the scope of this one to make other parts of the codebase more consistent.

Reply:
location is available in AssignMetadataMutator via `m.Path()`:

https://github.com/open-policy-agent/gatekeeper/blob/b1ebeb5713c42b5dcd4e89674e1e221b2e34dc15/pkg/mutation/mutators/assignmeta_mutator.go#L63

Reply:
That being said, this is just building off of the pre-existing logic.

Might be a micro-optimization or just historical cruft. IIRC mutator validation initially was a separate path from mutator creation, but that was cleaned up. This may be a dangling artifact of that.

Reply:
Done.